src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -391,14 +397,63 @@ private IAIState canGoPlanting(@NotNull final ScarecrowTileEntity currentField,
      *
      * @param position the position to check.
      * @param field    the field close to this position.
-     * @return true if should be hoed.
+     * @return position of hoeable surface or null if there is not one
      */
-    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final ScarecrowTileEntity field)
+    private BlockPos findHoeableSurface(@NotNull BlockPos position, @NotNull final ScarecrowTileEntity field)
     {
+        position = getSurfacePos(position);
+        if ((position == null)
+            || field.isNoPartOfField(world, position) 
+            || (world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
+            || (world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
+            || !world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT)
+        )
+            {
+                return null;
+            }
+        return position;
+    }
+
 
-        return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.above()).getBlock() instanceof CropsBlock)
-                 && !(world.getBlockState(position.above()).getBlock() instanceof BlockScarecrow)
-                 && (world.getBlockState(position).getBlock().is(Tags.Blocks.DIRT) || world.getBlockState(position).getBlock() instanceof GrassBlock);
+    private BlockPos getSurfacePos(final BlockPos position)
+    {
+        return getSurfacePos(position, 0);
+    }
+    /**
+    * Finds the position of the surface near the specified position
+    *
+    * @param position the location to begin the search
+    * @return the position of the surface block or null if it can't be found
+    */
+    private BlockPos getSurfacePos(final BlockPos position, final Integer depth)
+    {
+        if (depth > MAX_DEPTH)
+        {
+            return position.below();
```

Comment:
shouldn't this be null too?

Reply:
Actually... The intention here was that once it reached the max height the farmer would level the ground there to work, but you're right, I think it would be better if they just stopped. That would allow farmland to butt up against a cliff, too